File: src/main/kotlin/magicbook/gtlitecore/common/entity/explosive/EntityTaraniumCharge.java
package magicbook.gtlitecore.common.entity.explosive;

import gregtech.common.entities.EntityGTExplosive;
import magicbook.gtlitecore.common.block.GTLiteMetaBlocks;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.world.World;
import org.jetbrains.annotations.NotNull;

public class EntityTaraniumCharge extends EntityGTExplosive
{

    @SuppressWarnings("unused")
    public EntityTaraniumCharge(World world)
    {
        super(world);
    }

    public EntityTaraniumCharge(World world, double x, double y, double z, EntityLivingBase exploder)
    {
        super(world, x, y, z, exploder);
    }

    @Override
    protected float getStrength()
    {
        return 400.0F;
    }

    @Override
    public boolean dropsAllBlocks()
    {
        return false;
    }

    @Override
    protected int getRange()
    {
        return 256;
    }

    @NotNull
    @Override
    public IBlockState getExplosiveState()
    {
        return GTLiteMetaBlocks.TARANIUM_CHARGE.getDefaultState();
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/common/entity/explosive/EntityTaraniumCharge.kt
package gregtechlite.gtlitecore.common.entity.explosive

import gregtech.common.entities.EntityGTExplosive
import gregtechlite.gtlitecore.common.block.GTLiteBlocks
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.EntityLivingBase
import net.minecraft.world.World

class EntityTaraniumCharge(world: World?,
                           x: Double, y: Double, z: Double,
                           exploder: EntityLivingBase?) : EntityGTExplosive(world, x, y, z, exploder)
{

    constructor(world: World?) : this(world, 0.0, 0.0, 0.0, null)

    override fun getStrength() = 400.0f

    override fun dropsAllBlocks() = false

    override fun getRange() = 256

    override fun getExplosiveState(): IBlockState = GTLiteBlocks.TARANIUM_CHARGE.defaultState

}
